subsys/testsuite/ztest/src/ztest_new.c

```diff
@@ -553,23 +576,53 @@ void end_report(void)
 K_APPMEM_PARTITION_DEFINE(ztest_mem_partition);
 #endif
 
+int ztest_list_tests(void)
+{
+	struct ztest_suite_node *ptr;
+	struct ztest_unit_test *test = NULL;
+	int test_count = 0;
+
+	TC_PRINT("Listing tests (suite::test)\n");
```

Comment:
can we also get some sort of machine-readable and stable output? e.g., json?

or just one-per line with nothing extra?

Reply:
I think most of Zephyr tests use junit's output format. But upon thinking about this more, can we do this as a standalone feature in twister by parsing the .elf file?

Reply:
Updated to list just one per line.